creme/creme_core/gui/menu.py

```diff
@@ -239,6 +251,143 @@ def url(self):
         return get_url()
 
 
+class TemplateEntry(MenuEntry):
+    template_name = 'creme_core.gui.menu.TemplateEntry'
+
+    def get_template(self, context):
+        template_name = context['template_name']
+
+        if template_name is None:
+            raise ImproperlyConfigured(
+                "TemplateEntry requires either a definition of "
+                "'template_name' or an implementation of 'get_template_name()'"
+            )
+
+        return get_template(template_name)
+
+    def get_context(self, request):
+        label = self.render_label(request)
+        user = request['user']
+
+        try:
+            self.check_permissions(user)
+            permission_error = ''
+        except PermissionDenied as e:
+            permission_error = str(e)
+
+        return {
+            'label': label,
+            'permission_error': permission_error,
+            'template_name': self.template_name
+        }
+
+    def render(self, request):
+        context = self.get_context(request)
+        return self.get_template(context).render({
+            'user': request['user'],
+            'request': request,
+            'entry': context,
+        })
+
+
+class ActionEntry(TemplateEntry):
+    template_name = 'creme_core/menu/action.html'
+
+    form_class = menu_forms.ActionEntryForm
+
+    action: str = 'redirect'
+    url_name: str = ''
+
+    icon_name: str = None
+    icon_title: str = None
+    classes: Sequence[str] = ()
+    description: str = ''
+
+    @property
+    def url(self) -> str:
+        url_name = self.url_name
+        return reverse(url_name) if url_name else ''
+
+    def _get_prop(self, name, context):
+        prop = getattr(self, f'get_{name}', None)
+        return prop(context) if callable(prop) else getattr(self, name, None)
```

Comment:
I'm really not fan of this kind of code ; you (or or IDE...) have to read the code to know the API (and you do not have a cool place to write documentation or typing, so you do not write documentation/typing -- YOLO)
Write your 3 trivial methods like:

```
def get_icon_name(self, context) -> str:
   """Return the name of the used icon, like 'add' or 'edit'.""""
   return self.icon_name
```